### DISCUSSION TITLE
Option for jackc/pgx/stdlib driver for generated tests

### DISCUSSION THREAD

**Discussion by rajaiswal** [aarondl/sqlboiler#1235] [Ideas]
I am using SQLBoiler with `jackc/pgx/v5/stdlib` as the postgres driver. I was wondering if we could have a config option to use `jackc/pgx/v5/stdlib` as driver for generated tests instead of `lib/pq`? I tried it out by manually editing the generated test files and the tests seem to run good, but would like to see a permanent solution. Is that something which can be implemented easily?

**Comment by stephenafamo** [comment]
It isn't *too* complicated. but it just touches a lot of files. If pgx **also** used the same driver name as `pq`, we would only need to add a flag and dynamically change the import [here](https://github.com/volatiletech/sqlboiler/blob/master/drivers/sqlboiler-psql/driver/psql.go#L986).

However, because we also need to change the driver name when opening the connection, we also need to modify the template. And to do that....

A flag will need to be added to `main.go`, then read that config and pass it to `boilingcore.Config` and then pass it to `boilingcore.State`, then pass it to `boilingcore.templateData` then modify the template to use a different driver.

Also, this will add a flag to the entire sqlboiler CLI when only one driver needs it which I'm not so sure of.

If you're worried about having extra dependencies, then you shouldn't worry too much since a dependency used only in your test won't be compiled with the rest of your code. 


The alternative is to change this for **everyone**, but I'm not so sure we should do this since there are no real issues with using `lib/pq` at the moment.

**Reply by rajaiswal** [reply]
@stephenafamo Appreciate the detailed response. One advantage I noticed of using using `jackc/pgx/v5/stdlib` over `lib/pq` is that this (https://github.com/volatiletech/sqlboiler/issues/1234) problem goes away and all tests pass even if you are using the latest `ericlargergren/decimal` version `v0.0.0-20211103172832-aca2edc11f73` on SQLBoiler v4.14.0. Though not sure if this would make it a good enough reason to change it for **everyone**.

**Reply by stephenafamo** [reply]
That would be a good reason 😅

I'll consider it a bit more
